Cython/Compiler/Code.py

```diff
@@ -2378,8 +2388,10 @@ def put_error_if_unbound(self, pos, entry, in_nogil_context=False):
             self.globalstate.use_utility_code(
                 ExprNodes.raise_unbound_local_error_utility_code)
 
+        if not null_code:
+            null_code = entry.type.check_for_null_code(entry.cname)
```

Comment:
`null_code` sounds like we're overriding the code for `NULL` here, not a complete null check. Let's name it `unbound_check_code`, even if it's a bit longer. It's not the callers concern that "unbound" is defined as being `NULL`.